### PR TITLE
Add test suite coverage to deployment

### PR DESCRIPTION
**Pull request by snuggs**
Going off the awesome work we did in #164 we now have a fairly stable (and well tested) build pipeline.

🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉  #WOOT!!!! 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 

Coverage & Lint are on deck.

Thanks for setting the `bin/crank` foundation @tmornini @pachonk. NOW is where the rubber meets the road.

### CODE REVIEW

**Comment by snuggs** [comment]
https://travis-ci.org/devpunks/snuggsi/builds/402507678?utm_source=github_status&utm_medium=notification

HMMMM. Well the great news is tests work (TOO well). @tmornini @pachonk  any idea why the suite is running twice? I know the tests had been running before but perhaps shouldn't be in `bin/crank`? It works great for local because if any of the steps in `bin/crank` don't pass it disables deployment.

Thoughts... 🤔

**Comment by snuggs** [comment]
@brandondees @tmornini @pachonk #PING.

**Comment by brandondees** [comment]
`$ bin/test && bin/cover` does bin/cover run the tests again?

**Comment by rianby64** [comment]
Looks like it does, @brandondees

**Comment by snuggs** [comment]
@brandondees @rianby64 hmmm seems like `bin/cover` is commented tho. I was gonna address after this gets merged. I have no clue why it's running twice. Perhaps @tmornini made some setting somewhere in Travis.

**Comment by rianby64** [comment]
Check this out

https://github.com/devpunks/snuggsi/blob/352f820308e0f9e9b3900f5f4c5da38590f39411/.travis.yml#L20

Right in the ```.travis.yml``` you are calling twice the test process.
Just drop ```- bin/test``` and that will be enough, because ```- bin/cover``` fires the tests in fact.

**Comment by snuggs** [comment]
@rianby64 you nailed it! I would have never thought to look in there as it was @tmornini and @pachonk who worked on the integration. It feels like there should be a `bin/integrate` to go with our ontology.  @brandondees since i've known him has always reminded me developers should have well named "helpers" as well. Not just code. 

"integrate" an "integration" sounds like what we are doing here. Implementation details can be tucked in there. Either way feels better than having a conjunction for a "single" entry point nonetheless.

I'll "WIP" something up (pun intended) 😎

**Comment by brandondees** [comment]
@snuggs sometimes you can't get away from what people are already calling a thing. sometimes you can (but beware the confusion you'll cause).

binstubs are just scripts that can be executed directly: `bin/deploy` as opposed to `sh bin/deploy`

**Comment by tmornini** [comment]
Not a fan of putting all runables in bin/

Put all things in context, or it's Fake News, right?

If ```bin/test``` tests the entire system, it's context is /, therefore it really belongs in ```/test```

Which is to say ```bin/text``` ***should*** test everything in ```bin/```. 😄

**Comment by brandondees** [comment]
I don't disagree with the ideology, @tmornini, but running things from in `bin/*` is pretty widely established convention, mirroring unix, and there's some pragmatic benefits to rolling with that for the sake of convenience and adoption. We're in the process of standardizing a set of common command names that should be expected to work in any dev project regardless of the tech specifics (language, framework, architecture, etc)

**Comment by tmornini** [comment]
Any project that needs to work with snuggsi is going to need to execute the right commands.

It's irrelevant where those commands live.

You're welcome to put the commands into the wrong context if you like, of course -- just like you've welcome to write hard-to-read, buggy, inefficient code. After all, that's how most code is written so perhaps this project should follow those conventions as well. 😄

File: bin/snuggsi [hunk 1]
```diff
@@ -11,7 +11,10 @@ echo "  ツツツツツツツツツツツツツツツツツツツツツツツツ
 echo
 
 function canonical {
-  echo Canonical location for:
+  local what='foo'
+   nice='shazam'
```
Comment (tmornini) [inline bin/snuggsi]: indentation is incorrect 😄

**Comment by snuggs** [comment]
@tmornini @brandondees perhaps can have cake and eat it too? We do use this in package.json for this very reason so it's best to get this correct as then can just be an implementation detail.

https://docs.npmjs.com/files/package.json#bin

**Comment by snuggs** [comment]
@tmornini done and done! Bash scripting you'd be proud of. :-) 

/cc @rianby64 @brandondees

**Comment by brandondees** [comment]
@snuggs see, that wasn't so hard was it?

**Pull request merged**